api/server/httputils/errors.go

```diff
@@ -48,7 +49,19 @@ func GetHTTPErrorStatusCode(err error) int {
 		// there are errors falling back into this logic.
 		// If we need to differentiate between different possible error types,
 		// we should create appropriate error types that implement the httpStatusError interface.
+
 		errStr := strings.ToLower(errMsg)
+		// if the error has a type of grpc.rpcError, the error message should
+		// contain a sub string of "rpc error: code = "
+		if strings.Contains(errStr, "rpc error: code = ") {
```

Comment:
This string check is probably more expensive, and certainly more fragile, than just checking `statusCodeFromGRPCError` in all cases.

If we move this to the top of the `default` case, in many cases we won't have to mess around with this string handling.
so...

```go
default:
    statusCode = statusCodeFromGRPCError(err)
    if statusCode == http.StatusInternalServerError {
        // nasty string comparison stuff
    }
````